delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resolvers/ResolverResolutionRejection.scala

```diff
@@ -19,15 +19,42 @@ sealed abstract class ResolverResolutionRejection(val reason: String) extends Pr
 
 object ResolverResolutionRejection {
 
+  /**
+    * Rejection when the access to the given project has been denied by using the identity resolution
+    * defined in the resolver
+    * @param projectRef          the project where we attempted the resolution
+    * @param identityResolution  the identity resolution used
+    */
+  final case class ProjectAccessDenied(projectRef: ProjectRef, identityResolution: IdentityResolution)
+      extends ResolverResolutionRejection(
+        s"Access to project '$projectRef' has been denied using resolution $identityResolution"
+      )
+
+  /**
+    * Rejection returned when a resource has been found but has been filtered out because its types didn't match
+    * those declared in a cross-project resolver
+    * @param types the resource type
+    */
+  final case class ResourceTypesDenied(projectRef: ProjectRef, types: Set[Iri])
+      extends ResolverResolutionRejection(
+        s"None of the provided types '${types.mkString(",")}' matched the resolver authorized types"
+      )
+
+  /**
+    * Rejection when fetching the resource from the project failed
+    * @param reason a descriptive message as to why the rejection occurred
+    */
+  sealed abstract class FetchRejection(reason: String) extends ResolverResolutionRejection(reason)
+
   /**
     * Rejection returned when a subject intends to retrieve a resource using a resolver at a specific revision,
     * but the provided revision does not exist.
     *
     * @param provided the provided revision
     * @param current  the last known revision
     */
-  final case class RevisionNotFound(provided: Long, current: Long)
-      extends ResolverResolutionRejection(
+  final case class RevisionNotFound(projectRef: ProjectRef, provided: Long, current: Long)
```

Comment:
The ``projectRef`` wasn't there in order to keep the rejection as is on the rest of resources.
If you want to aggregate the rejections by project, it should probably be done in the resolution logic, not here